src/scripts/main.js

```diff
@@ -1,3 +1,67 @@
 'use strict';
 
 // write code here
+
+const rowAppended = document.querySelector('.append-row');
+const rowRemover = document.querySelector('.remove-row');
+const columnAppended = document.querySelector('.append-column');
+const columnRemover = document.querySelector('.remove-column');
+const field = document.querySelector('.field');
+
+function updateButtonStates() {
+  const trs = field.querySelectorAll('tr');
+  const tds = trs.length > 0 ? trs[0].children : [];
+
+  rowAppended.disabled = trs.length >= 10;
+  rowRemover.disabled = trs.length <= 2;
+  columnAppended.disabled = tds.length >= 10;
+  columnRemover.disabled = tds.length <= 2;
+}
+
+updateButtonStates();
+
+rowAppended.addEventListener('click', (e) => {
+  const trs = field.querySelectorAll('tr');
+
+  if (trs.length < 10) {
+    const newRow = field.insertRow();
+
+    for (let i = 0; i < trs[0].children.length; i++) {
+      newRow.insertCell();
+    }
+    updateButtonStates();
+  }
+});
+
+rowRemover.addEventListener('click', (e) => {
+  const trs = field.rows;
+
+  if (trs.length > 2) {
+    field.deleteRow(trs.length - 1);
+    updateButtonStates();
+  }
+});
+
+columnAppended.addEventListener('click', (e) => {
+  const trs = field.querySelectorAll('tr');
```

Comment:
Similar to the previous comment, consider using `field.rows` instead of `querySelectorAll('tr')` for better performance and readability.